content/7-webapp-features/1-fetch-data/solid/App.jsx

```diff
@@ -0,0 +1,12 @@
+import Users from './Users';
+import { Suspense, ErrorBoundary } from 'solid-js';
+
+export default function App() {
+	return (
+		<ErrorBoundary fallback={<p>An error occurred while fetching users</p>}>
+			<Suspense fallback={<p>Fetching users...</p>}>
+				<Users />
+			</Suspense>
+		</ErrorBoundary>
```

Comment:
If the app only contains `Users` component, why not.
But SolidJS `createResource` providing the `loading` and `error` keys is very interesting to showcase in a large app.
https://www.solidjs.com/tutorial/async_resources?solved

I think `Suspense` & `ErrorBoundary` would be interesting to showcase when there are multiple promise calls

Reply:
Oh, so this does not seem to be on the docs. It was my impression that an ErrorBoundary was required (since during my tests, `createResource` did not catch the error, getting an `uncaught promise` in the console). According to [this issue](https://github.com/solidjs/solid/issues/714) the error throw is deferred until you _read_ `data()`. So, as long as you don't attempt to read it if there's something in `data.error`, no error boundary is necessary.

Another thing is that not only the Error is deferred to the `data()` read, but also suspension. So both the `ErrorBoundary` and `Suspense` could be within the `User`  component, wrapping the `users()` call directly.

Regardless, I've updated the example showcasing `data.loading` and `data.error`. Let me know what you think!